lib/links/parser.rb

```diff
@@ -2,51 +2,58 @@
 
 class Parser
 
-    def initialize(url)
-        @elements = {a: ["href"],
-                img: ["src"],
-                script: ["src"],
-                frame: ["src"],
-                iframe: ["src"],
-                link: ["href"],
-                source: ["src", "srcset"],
-                track: ["src"] }
-
-        @url = url
+  def initialize(url)
+    @elements = { a: ['href'],
+                  img: ['src'],
+                  script: ['src'],
+                  frame: ['src'],
+                  iframe: ['src'],
+                  link: ['href'],
+                  source: ['src', 'srcset'],
+                  track: ['src'] }
+
+    @url = url
+  end
+
+  def run
+    all_links = []
+    @doc = Nokogiri::HTML(HTTParty.get(@url))
```

Comment:
Nitpicking a bit but how would you feel about extracting this to a `doc_html` function?

```ruby
def doc_html
  @doc ||= Nokogiri::HTML(HTTParty.get(@url))
end
```

See if you can understand why the `@doc ||=` part is there 😉

Reply:
Avoids the request if the variable is already set to something. Cool!